### PR TITLE
Add converter function for resource name constructor parameter

### PR DESCRIPTION
**Pull request by ni-erikcrank**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [ ] ~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~
- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Adds a converter function to convert a resource_name parameter that can specify multiple instruments and channels from a python sequence (list, tuple) to a comma-separated string as expected by the driver. This new function is currently not used but will soon be used by the nidcpower session constructor once it supports independent channels. 

### What testing has been done?

Added unit tests for the converter function in nifake/unit_tests/test_converters.py.

### CODE REVIEW

File: build/templates/_converters.py.mako [hunk 1]
```diff
@@ -274,6 +274,73 @@ def convert_chained_repeated_capability_to_parts(chained_repeated_capability):
...
+    # First we deal with a list
+    resource_name_list = resource_name.split(',')
+    if len(resource_name_list) > 1:
+        # We have a list so call ourselves again to let the iterable instance handle it
+        return _convert_resource_name(resource_name_list)
+
+    # Must be of form "Dev/0", "Dev/0:1", "Dev/0-1" or "Dev"
+    fields = [item.strip() for item in resource_name.split('/')]
+    device_name = fields[0]
+    if len(device_name) == 0:
```
Comment (ni-erikcrank) [inline build/templates/_converters.py.mako:318]: I was unclear whether this code should check for empty device name or if it should pass the empty device name to the driver to report the error.

**Comment by sbethur** [comment]
ok to test

File: build/templates/_converters.py.mako [hunk 2]
```diff
@@ -274,6 +274,73 @@ def convert_chained_repeated_capability_to_parts(chained_repeated_capability):
...
+# We cannot use collections.abc.Iterable here because strings are also iterable and then this
+# instance is what gets called instead of the string one.
+@_convert_resource_name.register(list)  # noqa: F811
+@_convert_resource_name.register(tuple)  # noqa: F811
+def _(resource_name):
+    '''Iterable version - can handle lists and tuples'''
+    resource_name_list = []
+    for r in resource_name:
+        resource_name_list += _convert_resource_name(r)
+    return resource_name_list
```
Comment (smooresni) [inline build/templates/_converters.py.mako:341]:
```suggestion
    return [_convert_resource_name(r) for r in resource_name]
```

Comment (smooresni) [reply build/templates/_converters.py.mako:341]: just a suggestion.

File: generated/nifake/nifake/unit_tests/test_converters.py [hunk 3]
```diff
@@ -237,6 +237,115 @@ def test_repeated_capabilies_without_prefix():
...
+    test_result_list = _converters.convert_resource_name(('Dev1', 'Dev2/0-2'))
+    assert test_result_list == 'Dev1,Dev2/0,Dev2/1,Dev2/2'
+
+
+def test_resource_name_mixed():
+    test_result_list = _converters.convert_resource_name(['Dev1', ('Dev2/0', 'Dev2/1'), ['Dev3/0:1', 'Dev4/1:2'], 'Dev5/1'])
+    assert test_result_list == 'Dev1,Dev2/0,Dev2/1,Dev3/0,Dev3/1,Dev4/1,Dev4/2,Dev5/1'
+
+
+def test_invalid_resource_name():
```
Comment (smooresni) [inline generated/nifake/nifake/unit_tests/test_converters.py:316]: Any reason not to use pytest.raises?
https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions

File: generated/nifake/nifake/unit_tests/test_converters.py [hunk 4]
```diff
@@ -237,6 +237,115 @@ def test_repeated_capabilies_without_prefix():
...
+    try:
+        _converters.convert_resource_name('')
+        assert False
+    except errors.InvalidResourceNameError:
+        pass
```
Comment (smooresni) [inline generated/nifake/nifake/unit_tests/test_converters.py:321]:
```suggestion
    with pytest.raises(errors.InvalidResourceNameError):
        _converters.convert_resource_name('')
```

**Comment by ni-erikcrank** [comment]
After discussing with Shreyas, we decided to modify the existing repeated capabilities converter instead.

**Pull request closed without merging**